packages/datagateway-download/src/downloadCart/downloadCartTable.component.test.tsx

```diff
@@ -180,8 +203,13 @@ describe('Download cart table component', () => {
     });
 
     expect(getDatafileCount).toHaveBeenCalled();
+    expect(wrapper.find('[aria-colindex=4]').find('p').first().text()).toEqual(
+      '7'
+    );
+
+    // TODO fix this. It seems getDatafileCount is not being called if entityType === 'datafile'
     expect(wrapper.find('p#fileCountDisplay').text()).toEqual(
-      expect.stringContaining('downloadCart.number_of_files: 28')
+      expect.stringContaining('downloadCart.number_of_files: 28 / 5000')
     );
```

Comment:
There's something weird going on here and I'm struggling to figure out what and why. The test is correct but failing - there are 4 cart items, `getDatafileCount` should return 7 for every item, so we should get 28 as the number of files. However, we are instead getting a result of 22 and it seems to be because whenever `entityType` === 'datafile', we return 1. But because `getDatafileCount` is being mocked, this 1 is not coming from this function. My guess is it's coming somewhere from `useDatafileCounts()`

```
// useDatafileCounts
return data
      ? data.map((cartItem) => {
          const { entityId, entityType } = cartItem;
          return {
            queryKey: ['datafileCount', entityId],
            queryFn: () =>
              datafileCountslimit(getDatafileCount, entityId, entityType, {
                apiUrl,
              }),
            onError: (error) => {
              handleICATError(error, false);
            },
            staleTime: Infinity,
            enabled: entityType !== 'datafile',
            initialData: entityType === 'datafile' ? 1 : undefined,        // Maybe to do with this?
          };
        })
      : [];
```
So for whatever reason, `getDatafileCount` is not being called if the `entityType` is a datafile. I'm having an ill day and am still to fully refamiliarise myself with React Query to figure out how to fix this at the moment but thought this would be a good note to raise.

Reply:
@sam-glendenning ah, this is my bad that I added this optimisation but didn't adjust the test for it. We don't need to send a request to count datafiles for a single datafile (as we know it's 1), and I figured the most efficient way to stop this is to disable the react-query request if `entityType === 'datafile'`, and we essentially "mock" the data by using `initialData` when the `entityType` is `datafile`. This ensures that the query function is not called for datafiles (thus better optimisation) but that the `data` prop of the query result is correctly `1`.

I'll fix the test to account for this behaviour

Reply:
I thought so! Nice bit of code nonetheless